### PR TITLE
use slices and map from the go stdlib instead of golang.org/x/exp

### PR DESCRIPTION
**Pull request by dastrobu**
**What this PR does**:

use slices and map from the go stdlib instead of golang.org/x/exp

**Which issue(s) this PR fixes**:

`nil`

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by dastrobu** [comment]
As we have go 1.23 now, I guess it would be nice to prefer the stdlib over `golang/x/exp` for maps and slices. 

I did not add this to the CHANGELOG yet, as I am not sure if you want this kind of cleanup mentioned there.

File: modules/overrides/runtime_config_overrides.go [hunk 1]
```diff
@@ -277,7 +278,7 @@ func (o *runtimeConfigOverridesManager) GetTenantIDs() []string {
 		return nil
 	}
 
-	return maps.Keys(tenantOverrides.TenantLimits)
+	return slices.Collect(maps.Keys(tenantOverrides.TenantLimits))
```
Comment (joe-elliott) [inline modules/overrides/runtime_config_overrides.go:281]: it seems like `slices.Collect` calls result in repeated appends to to an empty slice which is a bit unfortunate.

https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/slices/iter.go;l=49-54

wdyt about pre-allocing the slice to the appropriate capacity from the map and then using AppendSeq?

**Review by joe-elliott — COMMENTED** [review]
Thanks for the cleanup. Looking for your thoughts on a small perf improvement.

File: modules/overrides/runtime_config_overrides.go (hunk 1 above)
Comment (dastrobu) [reply modules/overrides/runtime_config_overrides.go]: not sure if this is going to be measurable, but why not. At least that should be closer to https://cs.opensource.google/go/x/exp/+/701f63a6:maps/maps.go;l=21

**Review by joe-elliott — APPROVED** [review]
Thanks!

**Pull request merged**